developer/frameworks/articles/data-engine/04-data-layouts.markdown

```diff
@@ -0,0 +1,162 @@
+# Data Layouts
+
+Data Layouts control the display of a from backed by the data engine. The basic
+functionality is to lay the field out in _rows_ and _columns_.
+
+## Adding Data Layouts
+
+Add data layouts by making a POST request to the `data-layouts` endpoint,
+passing in the ID of the data definition the data layout is to be created for:
+
+```sh
+curl -i -X POST \
+    http://localhost:8080/o/data-engine/v1.0/data-definitions/{data_definition_id}/data-layouts \
+    -H 'Authorization: Basic dGVzdEBsaWZlcmF5LmNvbTp0ZXN0' \
+    -H 'Content-Type: application/json' \ 
+{
+    "dataLayoutPages": [
+        {
+            "dataLayoutRows": [
+                {
+                    "dataLayoutColums": [
+                        {
+                            "columnSize": "1",
+                            "fieldNames": [
+                                "fieldName"
+                            ]
+                        }
+                    ]
+                }
+            ],
+            "description": {"en_US":"description"},
+            "title": {"en_US":"title"}
+        }
+    ],
+    "description": {"en_US":"description"},
+    "name": {"en_US":"name"},
+    "paginationMode": "single-page"
+}
+
+```
+
+Here you can create multiple pages, laying out your fields in column and rows.
+Each page can have a title and description, and the `dataLayout` itself can have
+a name and description (a name is required).
+
+## Retrieving Data Layouts
+
+### Retrieving all the Data Layouts for a Data Definition
+
+Send a GET request to the `data-definitions` resource's `data-layouts` endpoint,
+passing in the data definition ID as a URL parameter: 
+
+```sh
+curl -i -X GET \
+    http://localhost:8080/o/data-engine/v1.0/data-definitions/{data_definition_id}/data-layouts \
+    -H 'Authorization: Basic dGVzdEBsaWZlcmF5LmNvbTp0ZXN0' \
+    -H 'Content-Type: application/json' \ 
+```
+
+### Retrieving all the Data Layouts in a Site 
+
+Send a GET request to the `data-definitions` resource's `data-layouts` endpoint,
+passing in the data definition ID as a URL parameter: 
+
+```sh
+curl -i -X GET \
+    http://localhost:8080/o/data-engine/v1.0/data-definitions/{data_definition_id}/data-layouts \
+    -H 'Authorization: Basic dGVzdEBsaWZlcmF5LmNvbTp0ZXN0' \
+    -H 'Content-Type: application/json' \ 
+```
+
+You could use this call to get the data layout key for a particular data layout,
+then get the layout by this key.
+
+### Retrieving the Data Layout Matching a Key
+
+Send a GET request to the `data-layouts` endpoint, passing in both the
+`{site_id}` and the `{data_layout_key}` as URL parameters:
+
+```sh
+curl -i -X GET \
+    http://localhost:8080/o/data-engine/v1.0/sites/{site_id}/data-layouts/{data_layout_key} \
+    -H 'Authorization: Basic dGVzdEBsaWZlcmF5LmNvbTp0ZXN0' \
+    -H 'Content-Type: application/json' \ 
+```
+
+- `dataLayoutKey` is 
```

Comment:
Hay @marcelabc can you help me describe the `dataLayoutKey` parameter here?

Reply:
It's a string that can be pass in the request or not. If the user doesn't include this information in the request body, the system will generate automatically.

Reply:
Okay, so it's a String that a developer can add when adding a data layout, to provide their own backend identifier for the data layout? And then, they can use it here to retrieve  data layouts that match the key? Am I understanding correctly?
@marcelabc

Reply:
I don't know what happens if the user tries to add 2 data definition and put the same key in both of them.. 
I will investigate and I can tell you tomorrow, but I don't think the user creates this usually because this field is hidden in the frontend so the system always creates.
but it is better to know what happens if this eventually happens

Reply:
Yes, you understood it right!